File: Server/PartyService/PartyService.Web.Api/Models/Parties/CreatePartyRequestModel.cs
namespace PartyService.Web.Api.Models.Parties
{
    using System;
    using System.ComponentModel.DataAnnotations;

    public class CreatePartyRequestModel
    {
        [Required]
        public string Title { get; set; }

        [Required]
        public string Description { get; set; }

        [Required]
        public DateTime StartTime { get; set; }

        [Required]
        public double Longitude { get; set; }

        [Required]
        public double Latitude { get; set; }

        [Required]
        public string LocationAddress { get; set; }
    }
}

File: Client/IParty/IParty/CreatePartyRequestModel.m
//
//  CreatePartyRequestModel.m
//  IParty
//
//  Created by Swifty on 2/5/16.
//  Copyright © 2016 Swifty. All rights reserved.
//

#import "CreatePartyRequestModel.h"

@implementation CreatePartyRequestModel

-(instancetype)initWithTitle:(NSString *) title description:(NSString *)description locationAddress:(NSString *)locationAddress longitude:(NSNumber *) longitude latitude:(NSNumber *) latitude startTime:(NSString *) startTime {
    
    self = [super init];
    
    if(self) {
        self.title = title;
        self.pDescription = description;
        self.locationAddress = locationAddress;
        self.longitude = longitude;
        self.latitude = latitude;
        self.startTime = startTime;
    }
    
    return self;
}

+(JSONKeyMapper*)keyMapper {
    return [[JSONKeyMapper alloc] initWithDictionary:@{
                                                       @"description": @"pDescription"
                                                       }];
}

@end
